### PR TITLE
chore: setup frontpage and related components

### LINKED ISSUE #238: Create front page based on previous design
Follow up and create front page with Material UI.

### PR DESCRIPTION
**Pull request by progmatic-99**
Fixes #238

### CODE REVIEW

File: src/components/layout/footer.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <Box>
+      <Stack
+        alignContent="center"
+        justifyContent="center"
+        divider={<Divider orientation="vertical" flexItem />}
+        spacing={1}
+        direction={{ xs: 'column', md: 'row' }}
+      >
+        {footerLinks.map((linkArr) => {
+          return <FooterFields arr={linkArr} />
```
[the remark this answers is not in the document]
Comment (progmatic-99) [reply src/components/layout/footer.tsx:50]: Thanks, yeah, figured it out.

**Comment by progmatic-99** [comment]
Screenshot 2022-02-17 at 23-42-46 Screenshot

Screenshot 2022-02-17 at 23-43-26 Screenshot

**Comment by progmatic-99** [comment]
Screenshot 2022-02-18 at 22-59-40 Screenshot

**Comment by mariobehling** [comment]
- Upcoming events missing
- Sharing not working
- missing tags of events
- Create event button on top moving around
- Not showing event type, e.g. "Online Event"
- Use blue color same as on previous version, please
- Call for speakers not visible
- Groups not showing
- Use gray instead of white on footer
- Distribute footer more similarly to previous version
- No language switcher available
- header too big
- "Create event" on header not blue
- Font on header should not be bold
- "eventyay" on header should be further left (and not hardcoded)

**Comment by mariobehling** [comment]
Screenshot from 2022-02-19 16-02-30

**Comment by progmatic-99** [comment]
All of the above points, needs setting up different libraries. If I do them in this PR, it will become unnecessarily long. Future bugs would be hard to track.

**Comment by progmatic-99** [comment]
As, there is literally no-one using this site, we can take up these things step by step.

**Comment by mariobehling** [comment]
Please solve what can be solved without additional libraries. I do not think each single points requires a complete library., e.g. a very simple thing is that the header is too big and font bold. Please go through the list and fix as much as possible. If we would have opened issues for each single point in the past we would not have gotten as far as we are today.

**Comment by progmatic-99** [comment]
- [x] Upcoming events missing
- [x] "eventyay" on header should be further left
- [x] Not showing event type, e.g. "Online Event"
- [x] "Create event" on header not blue
- [x] Use gray instead of white on footer
- [x] Use blue color same as on previous version, please
- [x] Font on header should not be bold
- [x] header too big

Screenshot 2022-02-21 at 23-38-42 Screenshot

**Comment by progmatic-99** [comment]
Screenshot 2022-02-22 at 16-49-22 Screenshot

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
       - run: yarn install --immutable --immutable-cache --check-cache
       - run: yarn lint
       - run: yarn build
-      - run: yarn test
+      - run: yarn test --passWithNoTests
```
Comment (pc-beast) [inline .github/workflows/ci.yml:25]: why?

File: pages/index.tsx [hunk 3]
```diff
@@ -1,154 +1,86 @@
...
-        <title>Create Next App</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+export default function Index({ events, upcomingEvents, groups }): JSX.Element {
+  // const localTimezone = useTimezone((state) => state.localTimezone)
+  // const defaultTimezone = useTimezone((state) => state.defaultTimezone)
+  // const setTimezone = useTimezone((state) => state.setTimezone)
+  // const handleTimeZone = (e: any) => {
+  //   setTimezone(e.target.value)
+  // }
```
Comment (pc-beast) [inline pages/index.tsx:24]: We need to enable this for DateTime localization. Let's follow up on this in next PR.

File: pages/index.tsx [hunk 4]
```diff
@@ -1,154 +1,86 @@
...
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h3>
-              <Trans>Documentation</Trans> &rarr;
-            </h3>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+export async function getStaticProps() {
+  const date = new Date().toISOString()
+  const eventUrl = `https://api.eventyay.com/v1/events?cache=true
+  &filter=[{"and":[{"name":"state","op":"eq","val":"published"},{"name":"privacy","op":"eq","val":"public"},{"name":"is-featured","op":"eq","val":true}]},{"or":[{"name":"ends-at","op":"ge","val":"${date}"}]}]
```
Comment (pc-beast) [inline pages/index.tsx:60]: please refactor this as in ember js.

File: pages/index.tsx [hunk 5]
```diff
@@ -1,154 +1,86 @@
...
+  &page[size]=6
+  &public=true
+  &sort=starts-at`
+  const upcomingEventUrl = `https://api.eventyay.com/v1/events/upcoming?cache=true&page[size]=3&public=true&upcoming=true`
+  const groupsUrl = `https://api.eventyay.com/v1/groups?cache=true&filter=[{"name":"is-promoted","op":"eq","val":true}]&include=user,follower&page[size]=3&public=true`
```
Comment (pc-beast) [inline pages/index.tsx:65]: refactor the filters

File: pages/index.tsx [hunk 6]
```diff
@@ -1,154 +1,86 @@
...
 
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h3>
-              <Trans>Learn</Trans> &rarr;
-            </h3>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+  const [events, eventsErr] = await fetcher(eventUrl)
+  if (eventsErr) {
+    console.error(eventsErr)
```
Comment (pc-beast) [inline pages/index.tsx:69]: Let's setup MUI alerts  as reusable component in follow up PR for errors.

File: src/components/layout/footer.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import { Box, Container, Divider, Stack, Typography } from '@mui/material'
+import React from 'react'
+import Link from '../Link'
+import FooterFields from '../templates/FooterFields'
+
+const lastRow = [
+  'Maintained by Areeb Jamal & Team',
+  `© ${new Date().getFullYear()} eventyay`,
```
Comment (pc-beast) [inline src/components/layout/footer.tsx:8]: use ```dayjs``` to make it consistent. We have already setup ```dayjs``` in one of the earlier PRs

File: src/components/templates/event/EventCard.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  tz,
+  online,
+}): JSX.Element {
+  const options: Intl.DateTimeFormatOptions = {
+    dateStyle: 'full',
+    timeStyle: 'short',
+    timeZone: tz,
+  }
+
+  startsAt = new Intl.DateTimeFormat('en-US', options).format(
```
Comment (pc-beast) [inline src/components/templates/event/EventCard.tsx:26]: dayjs

File: src/components/templates/event/EventCard.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    new Date(startsAt)
+  )
+  endsAt = new Intl.DateTimeFormat('en-US', options).format(new Date(endsAt))
```
Comment (pc-beast) [inline src/components/templates/event/EventCard.tsx:29]: dayjs

File: src/utils/fetcher.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export default async function fetcher(url: string) {
+  try {
+    const data = await fetch(url).then((res) => res.json())
```
Comment (pc-beast) [inline src/utils/fetcher.ts:3]: Why to use async/await and .then at same line, both serve same purpose :confused:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (progmatic-99) [reply .github/workflows/ci.yml:25]: tests are empty. build error when pusing the code.

File: pages/index.tsx (hunk 5 above)
Comment (progmatic-99) [reply pages/index.tsx:65]: in what way?

File: src/utils/fetcher.ts (hunk 10 above)
Comment (progmatic-99) [reply src/utils/fetcher.ts:3]: it converts the data returned to serializable json format

**Comment by mariobehling** [comment]
Thanks. So, lets follow up about open points in the next PR.

**Pull request merged**